### PR TITLE
feat: Support setting TLS certificate lifetimes

### PR DESCRIPTION
**Pull request by razvan**
# Description

Part of: https://github.com/stackabletech/issues/issues/586

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

**Comment by sbernauer** [comment]
Just for the record: I reviews https://github.com/stackabletech/hbase-operator/pull/598, the comments also apply here

**Comment by sbernauer** [comment]
Just for the record for others: I'd say https://github.com/stackabletech/hbase-operator/pull/598 is the template - which is currently under review. Please place your feedback there

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Added
+
+- The lifetime of auto generated TLS certificates is now configurable with the role and roleGroup config property `requestedSecretLifetime` ([#660])
```
Comment (sbernauer) [inline CHANGELOG.md:9]:
```suggestion
- The lifetime of auto generated TLS certificates is now configurable with the role and roleGroup
  config property `requestedSecretLifetime`. This helps reducing frequent Pod restarts ([#660]).
```

File: rust/crd/src/lib.rs [hunk 2]
```diff
@@ -1191,6 +1237,10 @@ pub struct MiddleManagerConfig {
     /// [graceful shutdown documentation](DOCS_BASE_URL_PLACEHOLDER/druid/usage-guide/operations/graceful-shutdown).
     #[fragment_attrs(serde(default))]
     pub graceful_shutdown_timeout: Option<Duration>,
```
Comment (sbernauer) [inline rust/crd/src/lib.rs:1239]:
```suggestion
    pub graceful_shutdown_timeout: Option<Duration>,

```

File: rust/crd/src/lib.rs [hunk 3]
```diff
@@ -1148,6 +1189,10 @@ pub struct CoordinatorConfig {
...
```
Comment (sbernauer) [inline rust/crd/src/lib.rs:1191]:
```suggestion
    pub graceful_shutdown_timeout: Option<Duration>,

```

File: rust/crd/src/lib.rs [hunk 4]
```diff
@@ -1105,6 +1141,10 @@ pub struct BrokerConfig {
...
```
Comment (sbernauer) [inline rust/crd/src/lib.rs:1143]:
```suggestion
    pub graceful_shutdown_timeout: Option<Duration>,

```

File: rust/crd/src/lib.rs [hunk 5]
```diff
@@ -1234,6 +1285,10 @@ pub struct RouterConfig {
...
```
Comment (sbernauer) [inline rust/crd/src/lib.rs:1287]:
```suggestion
    pub graceful_shutdown_timeout: Option<Duration>,

```

File: rust/crd/src/lib.rs [hunk 6]
```diff
@@ -1277,6 +1333,10 @@ pub struct HistoricalConfig {
...
```
Comment (sbernauer) [inline rust/crd/src/lib.rs:1335]:
```suggestion
    pub graceful_shutdown_timeout: Option<Duration>,

```

File: rust/crd/src/security.rs [hunk 7]
```diff
@@ -190,6 +191,7 @@ impl DruidTlsSecurity {
         prepare: &mut ContainerBuilder,
         druid: &mut ContainerBuilder,
         pod: &mut PodBuilder,
+        requested_secret_lifetime: Duration,
```
Comment (sbernauer) [inline rust/crd/src/security.rs:194]: In all the other operators you passed it by reference ;) (remember to deref it in line 207)
```suggestion
        requested_secret_lifetime: &Duration,
```

File: rust/operator-binary/src/druid_controller.rs [hunk 8]
```diff
@@ -981,7 +981,12 @@ fn build_rolegroup_statefulset(
 
     // volume and volume mounts
     druid_tls_security
-        .add_tls_volume_and_volume_mounts(&mut cb_prepare, &mut cb_druid, &mut pb)
+        .add_tls_volume_and_volume_mounts(
+            &mut cb_prepare,
+            &mut cb_druid,
+            &mut pb,
+            merged_rolegroup_config.requested_secret_lifetime,
```
Comment (sbernauer) [inline rust/operator-binary/src/druid_controller.rs:988]:
```suggestion
            &merged_rolegroup_config.requested_secret_lifetime,
```

File: rust/crd/src/security.rs [hunk 9]
```diff
@@ -202,6 +204,7 @@ impl DruidTlsSecurity {
                             .with_node_scope()
                             .with_format(SecretFormat::TlsPkcs12)
                             .with_tls_pkcs12_password(TLS_STORE_PASSWORD)
+                            .with_auto_tls_cert_lifetime(requested_secret_lifetime)
```
Comment (sbernauer) [inline rust/crd/src/security.rs:207]:
```suggestion
                            .with_auto_tls_cert_lifetime(*requested_secret_lifetime)
```

**Review by sbernauer — CHANGES_REQUESTED** [review]
Only nits

**Review by sbernauer — APPROVED** [review]
Thanks! Let's hope it compiles ^^

**Comment by lfrancke** [comment]
Will this be in 24.11.1? It's a new feature so I guess it shouldn't but I'm not sure if we're set up with release branches?

**Pull request merged**